node/subsystem-util/src/rolling_session_window.rs

```diff
@@ -38,9 +40,18 @@ pub enum SessionsUnavailableReason {
 	/// The runtime API itself returned an error.
 	#[error(transparent)]
 	RuntimeApi(#[from] RuntimeApiError),
+	/// The chain API itself returned an error.
+	#[error(transparent)]
+	ChainApi(#[from] ChainApiError),
 	/// Missing session info from runtime API for given `SessionIndex`.
 	#[error("Missing session index {0:?}")]
 	Missing(SessionIndex),
+	/// Missing last finalized block number.
+	#[error("Missing last finalized block number")]
+	MissingLastFinalizedBlock,
+	/// Missing last finalized block hash.
+	#[error("Missing last finalized block hash")]
+	MissingLastFinalizedBlockHash,
```

Comment:
Maybe this error variant should have a parameter of a type `BlockNumber` to specify which block is missing?

Reply:
Good idea.